google/auth/external_account.py

```diff
@@ -523,6 +531,31 @@ def _create_default_metrics_options(self):
 
         return metrics_options
 
+    def _should_add_mtls(self):
+        """Returns a boolean representing whether the current credential is configured
+        for mTLS and should add a certificate to the outgoing calls to the sts and service
+        account impersonation endpoint.
+
+        Returns:
+            bool: True if the credential is configured for mTLS, False if it is not.
+        """
+        return False
+
+    def _get_mtls_cert_and_key_location(self):
+        """Gets the file locations for a certificate and private key file
+        to be used for configuring mTLS for the sts and service account
+        impersonation calls. Currently only expected to return a value when using
+        X509 workload identity federation.
+
+        Returns:
+            Tuple[str, str]: The cert and key file locations as strings in a tuple.
```

Comment:
I think this can be a follow up but I think in the future we may want to refactor this and the transport layer to wrap the cert and key as new sources get added.